src/sentry/snuba/metrics/query_builder.py

```diff
@@ -189,9 +190,33 @@ def parse_query(query_string: str) -> Sequence[Condition]:
             },
         )
         where, _ = query_builder.resolve_conditions(query_string, use_aggregate_conditions=True)
+
+        # XXX(ahmed): Hack to accept project:slug filter as we are no longer maintaining the
+        # metrics API.
+        # Essentially prior to this change `project` slug query filters were treated as `tags[
+        # project]` filters, so I loop over the where list and when its found it popped out of
+        # the list and replaced with the appropriate `project_id` filter.
+        for idx, condition in enumerate(list(where)):
```

Comment:
```suggestion
        for idx, condition in enumerate(where):
```

```
In [1]: gen = (1 for _ in range(5))

In [2]: gen
Out[2]: <generator object <genexpr> at 0x107efbb30>

In [3]: for idx, condition in enumerate(gen):
   ...:     print(idx, condition)
   ...:
0 1
1 1
2 1
3 1
4 1
```

Reply:
I am deleting elements from the list and I don't want to invalidate the iterator as I am doing so which is why I am using `list` to create a copy of that list 
As per this [SO](https://stackoverflow.com/a/23015166) answer, enumerate does not create a copy of the list